assets/js/vue-apps/form-components.js

```diff
@@ -17,8 +19,108 @@ function initCharacterCount() {
     }
 }
 
+function initBudgetInput() {
+    const mountEl = document.getElementById('js-vue-budget');
+    if (!mountEl) {
+        return;
+    }
+
+    new Vue({
+        el: mountEl,
+        delimiters: ['<%', '%>'],
+        data() {
+            return {
+                maxItems: 10,
+                maxBudget: 10000,
+                items: [],
+                fieldName: null,
+                ready: false,
+                error: null
+            };
+        },
+        watch: {
+            items: {
+                handler() {
+                    const lastItem = this.items[this.items.length - 1];
+                    if (lastItem) {
+                        if (this.items.length < this.maxItems && (!isEmpty(lastItem.item) || !isEmpty(lastItem.cost))) {
```

Comment:
The thinking here by @LynseyJReynolds was that it should automatically add a new row as soon as you populate either field in the row above.

Reply:
Makes sense. I'd suggest extracting this out into a method within component as this logic could get a little complex.